e2e/data-factory/persistence/adapters/ghost-api.ts

```diff
@@ -0,0 +1,31 @@
+import {ApiPersistenceAdapter} from './api';
+import type {BrowserContext} from '@playwright/test';
+
+type GhostApiResponse = {
+    [key: string]: unknown[] | unknown;
+};
+
+/**
+ * Ghost Admin API adapter that handles Ghost's API formatting conventions
+ * All Ghost Admin API endpoints follow the pattern:
+ * - Endpoint: /ghost/api/admin/{resource}
+ * - Request wrapper: { [resource]: [data] }
+ * - Response extractor: response[resource][0]
+ */
+export class GhostAdminApiAdapter extends ApiPersistenceAdapter<unknown, GhostApiResponse> {
+    constructor(context: BrowserContext, resourcePath: string, queryParams?: Record<string, string>) {
+        // Extract the resource name from the path (e.g., 'posts' from 'posts' or 'posts/1234')
+        const resource = resourcePath.split('/')[0];
+        
+        super({
+            context,
+            endpoint: `/ghost/api/admin/${resourcePath}`,
+            queryParams: queryParams || {},
+            wrapRequest: data => ({[resource]: [data]}),
```

Comment:
I think we should use a different name here, we are not wrapping request, but transforming data?